tests/streaming/test_dataloader.py

```diff
@@ -202,3 +205,46 @@ def test_dataloader_no_workers(tmpdir):
     assert len(dataset) == 1000
     assert len(dataloader) == 1000
     assert len(dataset) == 1000
+
+
+@pytest.mark.timeout(120)
+def test_dataloader_with_loading_states(tmpdir):
```

Comment:
Could you add a test with persistent workers ?

Reply:
Sure